src/pages/Contests/Contests.js

```diff
@@ -0,0 +1,307 @@
+    queryDefault.mentor ? queryDefault.mentor : false
+  )
+  const [mentorrCount, setMentorrCount] = useState(
+    queryDefault.mentor ? true : false
+  )
+
+  const [difficultyQueries, setDifficultyQueries] = useState(
+    queryDefault.divs ? queryDefault.divs.split(',') : []
+  )
+
+  const [isDiffChange, setIsDiffChange] = useState(
+    queryDefault.divs ? true : false
+  )
+
+  //Function to change the difficulty
+  const changeDifficultyFilter = (event, lev) => {
+    const res = event.target.checked
+    setIsDiffChange(true)
+    const difficultyAdd = difficultyLevels[lev]
+    if (res) {
+      setDifficultyQueries([...difficultyQueries, difficultyAdd])
+      console.log(difficultyQueries)
+      setDisplayDiff(
+        update(displayDiff, {
+          values: {
+            [lev]: {
+              $set: true,
+            },
+          },
+        })
+      )
+    } else {
+      const newList = difficultyQueries.filter(
+        (item) => item != difficultyFilters[lev]
+      )
+      setDifficultyQueries(newList)
+      setDisplayDiff(
+        update(displayDiff, {
+          values: {
+            [lev]: {
+              $set: false,
+            },
+          },
+        })
+      )
+    }
+  }
+  const handleSubmit = (e) => {
+    e.preventDefault()
+
+    var urlTo = '/contests?'
+
+    if (isDiffChange && difficultyQueries.length > 0) {
+      urlTo =
+        urlTo +
+        `divs=${JSON.stringify(difficultyQueries)
+          .replace(/"/g, '')
+          .replace(/]|[[]/g, '')}`
+    }
+    console.log(urlTo)
+
+    if (gymCount) {
+      urlTo = urlTo + `;gym=${JSON.stringify(gym)}`
+    }
+
+    console.log(urlTo)
+
+    if (mentorrCount) {
+      urlTo = urlTo + `;mentor=${JSON.stringify(mentorr)}`
+    }
+    console.log(urlTo)
+
+    window.location.href = urlTo
+  }
+
+  //searching for a contest
+  const handleSearch = (e) => {
+    // e.preventDefault()
+    const searchUrl = `/contests/?divs=${searchText}`
+    window.location.href = searchUrl
+  }
+
+  //opening filter block
+  function openNav() {
+    document.getElementById('mySidenav1').classList.remove('sideNavClose')
+    document.getElementById('mySidenav1').classList.add('sideNavOpen')
+    document.getElementById('contests_page_id').classList.add('blur_bckgrnd')
+    document.getElementById('page_heading_id').classList.add('blur_bckgrnd')
+    document.getElementById('contest_nav_id').style.display = 'none'
+  }
+
+  //closing filter block
+  function closeNav() {
+    document.getElementById('mySidenav1').classList.remove('sideNavOpen')
+    document.getElementById('mySidenav1').classList.add('sideNavClose')
+    document.getElementById('contests_page_id').classList.remove('blur_bckgrnd')
+    document.getElementById('page_heading_id').classList.remove('blur_bckgrnd')
+    document.getElementById('contest_nav_id').style.display = 'block'
+  }
+
+  useEffect(async () => {
+    await fetch(`https://api.codedigger.tech/contest/${queryStr}`, {
```

Comment:
Create an `/actions` folder in the `/src` directory and create a file named `/Contest.js`. Add the api call in that file as a function and import it here.

Reply:
Don't use `fetch`, use `axios`.